File: Matrices_Cs/src/Utilities/ConsoleInput.cs
using System;

namespace Matrices_Cs.Utilities
{
    public static class ConsoleInput
    {
        public static int GetInteger()
        {
            while (true)
            {
                var line = Console.ReadLine();
                if (int.TryParse(line, out int value)) return value;
                Console.Write("Entrada inválida, intente de nuevo: ");
            }
        }
    }
}

File: Matrices_Java/src/com/matrices/io/ConsoleInput.java
package com.matrices.io;

import java.util.Scanner;

public class ConsoleInput {
    private static final Scanner SC = new Scanner(System.in);

    public static int getInteger(String prompt) {
        while (true) {
            System.out.print(prompt);
            String line = SC.nextLine();
            try {
                return Integer.parseInt(line.trim());
            } catch (NumberFormatException e) {
                System.out.println("Entrada inválida, intente de nuevo.");
            }
        }
    }

    public static String getString(String prompt) {
        System.out.print(prompt);
        return SC.nextLine();
    }
}